python/py-findpython/Portfile

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-findpython
+version             0.6.2
+revision            0
+categories-append   devel
+platforms           {darwin any}
+license             MIT
+supported_archs     noarch
+
+python.versions     39 310 311 312 313
+
+maintainers         {gmail.com:davidgilman1 @dgilman} openmaintainer
+
+description         A utility to find python versions on your system.
+
+long_description    Also lets you filter for specific versions.
+
+homepage            https://github.com/frostming/findpython
+
+checksums           rmd160  19d4318f7511280d3f838e1b23e6a53ba73e92ad \
+                    sha256  e0c75ba9f35a7f9bb4423eb31bd17358cccf15761b6837317719177aeff46723 \
+                    size    17770
+
+python.pep517_backend pdm
+
+# borrowed from https://trac.macports.org/ticket/72050
+if {${name} ne ${subport}} {
+    if {${python.version} eq 39} {
+        build.args          --skip-dependency-check
```

Comment:
why do we need this?

Reply:
I was hitting the same dependency issue with pdm as seen in that ticket. You can also see it in the GH actions logs for the first commits I pushed to this PR, e.g. https://github.com/macports/macports-ports/actions/runs/13361861384/job/37312763006

It is only on python 3.9. I would drop 3.9 as a subport but poetry still will want it.

Reply:
looks like you're [missing a dependency on `py-packaging`](https://github.com/frostming/findpython/blob/bab436668eae84da79ca69d1f281fefb98b2a9a6/pyproject.toml#L8). Not sure where the message about missing `pdm-backend` and `importlib-metadata` originate from but just skipping the dependency check doesn't seem like a good idea....

Reply:
Take a look at the GH action output under "Installing dependencies for py39-findpython", you can see it is installing py-packaging. I was able to reproduce this locally too, where I could confirm that py39-packaging was installed but it would still fail with that error.

Reply:
you are not declaring a dependency on `py-packaging`, so it can't/shouldn't be installing that. I am not sure what the `pdm` backend is doing with the `pdm.lock` file - perhaps it's getting dependencies from there. But either way, this isn't correct.

Reply:
i'll do a push with suggested fixes and we'll see if it fails